src/addrman.cpp

```diff
@@ -77,8 +77,9 @@ double CAddrInfo::GetChance(int64_t nNow) const
     return fChance;
 }
 
-CAddrMan::CAddrMan(bool deterministic, int32_t consistency_check_ratio)
-    : insecure_rand{deterministic}
+CAddrMan::CAddrMan(std::vector<bool> asmap, bool deterministic, int32_t consistency_check_ratio)
+    : m_asmap{std::move(asmap)}
```

Comment:
question about use of `std::move` here - 

If reading `peers.dat` causes issues, `init.cpp` will instantiate a `CAddrMan` twice, using the same asmap both times. Is this ok? I know that `std::move` is more of a suggestion than a command, so I'm wondering how this code would be interpreted.

Reply:
The variable from `init.cpp` will be copied to the `CAddrMan` constructor into a temporary, used for the argument (this temporary is what is available inside the body of the constructor and will be automatically destroyed when the constructor returns). Then the constructor moves that temporary into the `m_asmap` member. So, the caller's variable from `init.cpp` will not be modified and can be reused.

See also https://github.com/bitcoin/bitcoin/pull/22791#discussion_r697472059

Reply:
hm interesting, I've made some little toy programs to try to understand this behavior, and I see that the caller's variable can be reused, so that resolves my concern.

I'm not seeing the behavior of copying the temporary, but I'm going to resolve this conversation and explain further on the comment you linked.